pkg/virt-controller/services/template.go

```diff
@@ -558,6 +560,33 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, i
 
 	setNodeAffinityForPod(vmi, &pod)
 
+	tscFrequency := nsr.TSCFrequency()
```

Comment:
How about adding a RenderAffinity() method instead, and encapsulating all of the logic there? I feel that fetching the TSC frequency from the NSR and then doing the calculation from the outside is a bit of an abuse. It would both be easier to understand and would allow us to extend affinity rendering in the future. WDYT?

Reply:
I would prefer the second option; additionally, we can simply stop setting the tscfrequency in the nodeselectorrenderer since it is not used there anymore.
Future enhancement could be to create a `podaffinityrenderer`